modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFRequestForm/utils/submitForm.ts

```diff
@@ -69,7 +71,24 @@ export default async function submitForm(
 
 	if (values?.activities?.length && dtoMDFRequest?.id) {
 		const dtoMDFRequestActivities = await Promise.all(
-			values?.activities?.map((activity) => {
+			values?.activities?.map(async (activity) => {
+
+				if (activity.onRemove && activity.id) {
+					if (activity.externalReferenceCodeSF) {
+						await deleteMDFRequestActivitiesSF(
+							ResourceName.ACTIVITY_SALESFORCE,
+							activity.externalReferenceCodeSF as string
+						);
+					}
+
+					await deleteMDFRequestActivities(
+						ResourceName.ACTIVITY_DXP,
+						activity.id as number
+					);
+
+					return null;
```

Comment:
it isnt necessary to filter if we have a dtoMDFRequestActivities with a null element?